src/odemis/odemisd/modelgen.py

```diff
@@ -277,6 +278,27 @@ def __init__(self, inst_file, settings_file=None, container=None, create_sub_con
 
         # TODO: check there is no cyclic dependencies on the parents/children
 
+    def _check_structure(self):
+        """
+        The AST should be a dict[str, dict[str, ...]]: name of component -> definition
+        Each component definition must have at least the "role" key
+        """
+        if not isinstance(self.ast, dict):
+            raise ParseError(f"Microscope file must be a dictionary of component names -> definition, but got a {type(self.ast).__name__}")
```

Comment:
Line is a bit too long
```suggestion
            raise ParseError(f"Microscope file must be a dictionary of component names -> definition, "
                             f"but got a {type(self.ast).__name__}")
```